src/backend/distributed/relay/relay_event_utility.c

```diff
@@ -497,7 +498,8 @@ TypeDropIndexConstraint(const AlterTableCmd *command,
 		{
 			/* we found the constraint, now check if it is for an index */
 			if (constraintForm->contype == CONSTRAINT_PRIMARY ||
-				constraintForm->contype == CONSTRAINT_UNIQUE)
+				constraintForm->contype == CONSTRAINT_UNIQUE ||
+				constraintForm->contype == CONSTRAINT_EXCLUSION)
```

Comment:
We have this kind of check twice in the codebase. I think we could extract them in a function and avoid the duplicate logic.